src/org/pentaho/reporting/platform/plugin/async/IAsyncReportState.java

```diff
@@ -29,6 +29,24 @@
   int getProgress();
 
   /**
+   *
+   * @return Page is currently being processed
+   */
+  int getPage();
+
+  /**
+   *
+   * @return Row is currently being processed
+   */
+  int getRow();
+
```

Comment:
`this.progress = (int) ReportProgressEvent.computePercentageComplete( event, true );`

It's already there. I will keep row in json as far as it is present on [mockup ](http://jira.pentaho.com/secure/attachment/64522/biserver-1225%20ux.png)